### PR TITLE
OCPBUGS-12730: Linuxptp failing to create pod node name label

### PR DESCRIPTION
**Pull request by edcdavid**
Some clusters are slow to configure cluster access right for the pod. If the linuxptp-daemon pod fails to access pods in the cluster, fail the linux ptpdaemon pod to trigger a pod recreation and retry.
In case of slow cluster, and if not retrying, the clock fails to acquire sync complaining about ptp:
```
E0419 21:08:35.036983 3962510 daemon.go:443] error trying to connect to event socket
```
initial unauthorized error:
```
E0419 21:34:50.318916 4020630 main.go:164] Unauthorized
E0419 21:34:50.344475 4020630 ptpdev.go:43] failed to get NodePtpDevice CR for node master1: Unauthorized
```

### CODE REVIEW

**Review by aneeshkp — APPROVED** [review]
/lgtm

File: cmd/main.go [hunk 1]
```diff
@@ -98,8 +98,11 @@ func main() {
 	}
 
 	// label the current linux-ptp-daemon pod with a nodeName label
-	labelPod(kubeClient, nodeName, podName)
-
+	err = labelPod(kubeClient, nodeName, podName)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply cmd/main.go:102]: yes, the return here will trigger a container restart until it succeeds

**Comment by aneeshkp** [comment]
/retitle [OCPBUGS-12730] Linuxptp: failing to create pod node name label

**Comment by aneeshkp** [comment]
/retitle OCPBUGS-12730: Linuxptp failing to create pod node name label

**Pull request merged**